twisted/web/test/test_http2.py

```diff
@@ -1403,6 +1403,32 @@ def validate(streamID):
         return a._streamCleanupCallbacks[1].addCallback(validate)
 
 
+    def test_resetAfterBody(self):
+        """
+        A client that immediately resets after sending the body causes Twisted
+        to send no response.
+        """
+        f = FrameFactory()
+        transport = StringTransport()
+        a = H2Connection()
+        a.requestFactory = DummyHTTPHandler
+
+        requestBytes = f.preamble()
+        requestBytes += buildRequestBytes(self.getRequestHeaders, [], f)
+        requestBytes += hyperframe.frame.RstStreamFrame(
+            stream_id=1
+        ).serialize()
+        a.makeConnection(transport)
+        a.dataReceived(requestBytes)
+
+        buffer = FrameBuffer()
+        buffer.receiveData(transport.value())
+        frames = list(buffer)
+
+        self.assertEqual(len(frames), 1)
```

Comment:
It's a SETTINGS frame. The server has to send one at the start of the connection: RFC 7540 also says this.